### PR TITLE
CDAP-2169: Delete the app if the last adapter got deleted

### PR DESCRIPTION
**Pull request by rohitsinha54**
Build: http://builds.cask.co/browse/CDAP-DT30-1

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java [hunk 1]
```diff
@@ -301,12 +302,24 @@ public void removeAdapter(Id.Namespace namespace, String adapterName)
...
-    // TODO: Delete the application if this is the last adapter
+    // delete the template if this was the last adapter of that type
+    deleteApp(Id.Application.from(namespace, adapterSpec.getTemplate()));
+  }
+
+  /**
+   * deletes the app if it has no associated adapters
+   * @param templateId the {@link Id.Application} of the template
+   */
+  private void deleteApp(Id.Application templateId) {
```
Comment (gokulavasan) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java]: Not sure if this is enough. Take a look at the AppLifecycleHttpHandler and see what all we when we delete an app. Shouldn't we cover all that?

**Comment by rohitsinha54** [comment]
Develop build : http://builds.cask.co/browse/CDAP-DUT1513-1

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java (hunk 1 above)
Comment (rohitsinha54) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java]: Yes. Make sense.  Will do that.

**Comment by rohitsinha54** [comment]
New PR: https://github.com/caskdata/cdap/pull/2204

**Pull request closed without merging**